workflow/src/test/java/com/synopsys/integration/alert/workflow/message/mocks/MockNotificationAccessor.java

```diff
@@ -37,8 +38,24 @@ public MockNotificationAccessor(List<AlertNotificationModel> alertNotificationMo
     }
 
     @Override
-    public List<AlertNotificationModel> findByCreatedAtBetween(OffsetDateTime startDate, OffsetDateTime endDate) {
-        return null;
+    public AlertPagedModel<AlertNotificationModel> findByCreatedAtBetween(OffsetDateTime startDate, OffsetDateTime endDate, int pageNumber, int pageSize) {
+        ArrayList<AlertNotificationModel> notifications = new ArrayList<>();
+        Predicate<AlertNotificationModel> beforePredicate = notification -> notification.getCreatedAt().isBefore(endDate) || notification.getCreatedAt().isEqual(endDate);
+        Predicate<AlertNotificationModel> afterPredicate = notification -> notification.getCreatedAt().isAfter(startDate) || notification.getCreatedAt().isEqual(startDate);
+        Predicate<AlertNotificationModel> withinRange = beforePredicate.and(afterPredicate);
+        for (AlertNotificationModel notification : alertNotificationModels) {
+            if (withinRange.test(notification)) {
+                notifications.add(notification);
+            }
+        }
```

Comment:
This can be streamed I believe. _alertNotificationModels.stream().map(withinRange::test).collect(Collectors.toList())_ which should remove the need to create the list on line 42

Reply:
yes you're right